src/main/java/pl/widulinski/webcrawlertool/urls/URLController.java

```diff
@@ -0,0 +1,100 @@
+package pl.widulinski.webcrawlertool.urls;
+
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.propertyeditors.StringTrimmerEditor;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
+import pl.widulinski.webcrawlertool.enums.Categories;
+import pl.widulinski.webcrawlertool.searchData.DataToSearch;
+import pl.widulinski.webcrawlertool.searchData.PreparedDataToSrchDto;
+import pl.widulinski.webcrawlertool.searchData.SearchDataService;
+import pl.widulinski.webcrawlertool.webDataToScrap.DataToScrapDto;
+import pl.widulinski.webcrawlertool.webDataToScrap.DataToScrapService;
+
+import javax.validation.Valid;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
+@Controller
+public class URLController {
+
+
+    @Autowired
+    private DataToScrapService dataToScrapService;
+
+    @Autowired
+    private SearchDataService searchDataService;
+
+    @InitBinder
+    public void initBinder(WebDataBinder binder) {
+        binder.registerCustomEditor(String.class, new StringTrimmerEditor(true));
+    }
+
+
+    @GetMapping("/adminMenu")
+    public String addShop(Model model) {
+        model.addAttribute("data", new DataToScrapDto());
```

Comment:
tu masz wyjańsnienie czemu DTO nie powinno być komponentem. Czasami tworzysz je samemu. component to klasa ktora całkowicie zarzadza spring. Pozatym domyślnie `@Component` tworzy Singletona czyli 1 instancje obiektu. A DTO czy model jest wielokrotnie tworzony w aplikacji